src/grading.asm

```diff
@@ -861,4 +875,329 @@ UpdateGradeSHIR:
+    cp a, 0 ; If it's 0, we don't need to do weird math.
+    jr z, .GetOffset
+    ld d, a ; ld d, 3
+    ld b, 5 
+    ld a, b ; ld a, 5
+    dec d ;dec 3 to 2, so we don't accidentally add more than intended
+:   add a, b ; 5+5 = 10 ; 10+5 = 15
+    dec d 
+    jr nz, :- ; go back if d isn't 0
+    ld b, a ; ld b, 15
+.GetOffset
+    ld a, [hLineClearCt]
+    cp a, 0 ; If no lines were cleared, we don't need to do anything, just continue
+    jp z, .IncreaseInternalGrade
+    add a, b
+    ld b, 0
+    ld c, a
+    add hl, bc
+    ld a, [hl]
+    ld e, a ; We will use almost all registers to get the multipliers, so we need to keep the points we should add in a safe spot 
+    jp .multipliers
+.loadpoints    
+    ld hl, wInternalGradePoints
+    add a, [hl]
+    ld [wInternalGradePoints], a
+    jp .IncreaseInternalGrade
+.multipliers
+    ; There are some multipliers to help us increase our grade faster
+    ld hl, sTGM3ComboMultipliers
+    ld a, [hComboCt] ; Example: 3
+    cp a, 0
+    
+    ld d, a ; ld d, 3
+    ld b, 5 
+    ld a, b ; ld a, 5
+    dec d ;dec 3 to 2, so we don't accidentally add more than intended
+:   add a, b ; 5+5 = 10 ; 10+5 = 15
+    dec d 
+    jr nz, :- ; go back if d isn't 0
+    ld b, a ; ld b, 15
+    ld a, [hLineClearCt]
+    cp a, 0 ; If no lines were cleared, we don't need to do anything, just continue
+    jr z, .levelmultiplier
+    add a, b
+    ld b, 0
+    ld c, a
+    add hl, bc
+    ld a, [hl] ; Now we got our multiplier!, let's apply it.
+    dec a ; A multiplier of 1 shouldn't change anything, so let's get rid of them
+    cp a, 0
+    jr z, .levelmultiplier; Continue
+    ld b, a ; Load the multiplier into B
+    ld a, e ; Remember the points we got earlier?, here they are
+    ld c, a ; We will add from C
+:   add a, c
+    dec b
+    jr nz, :-
+    ; Finally!, we can now load the points, right?, NO!, there is yet another multiplier...
+    ; We have to keep the points safe again...
+    ld e, a
+.levelmultiplier
+    ; Make HL point to the table that contains the level multipliers
+    ld hl, sTGM3LevelMultiplier
+    ; Get our level into BC
+    ld a, [hCLevel+3]
+    ld b, a
+    ld a, [hCLevel+2]
+    swap a
+    or b
+    ld c, a
+    ld a, [hCLevel+1]
+    ld b, a
+    ld a, [hCLevel]
+    swap a
+    or b
+    ld b, a
+.Level750
+    ; Is our level 750 or higher?
+    ld a, b
+    cp a, LEVEL_MULT_3A
+    ; If B is less than 7, that means we are not even in level 700, so check for 500
+    jr c, .Level500
+    ; If B is NOT less than 7, we might be in level 750 or greater, so check the remaining digits.
+    ld a, c
+    cp a, LEVEL_MULT_3B
+    ; If C is less than 50, we didn't reach level 750 yet, but we are for sure in the 7** Section, load the corresponding offset.
+    jp .under750
+    ; If C is equal or greater than 50, then congrats!, load the corresponding offset
+    ld b, 0
+    ld c, 2 
+    add hl, bc
+    ld a, [hl]
+    jp .Multiply
+.under750

```

Comment:
You seem to choose .UpperCase and .lowercase almost randomly. Best to choose one and stick with it.